operator/v1/types_network.go

```diff
@@ -533,6 +536,15 @@ type ProxyConfig struct {
 	ProxyArguments map[string]ProxyArgumentList `json:"proxyArguments,omitempty"`
 }
 
+// EgressIPConfig defines the configuration knobs for egressip
+type EgressIPConfig struct {
+	// EgressIPReachabilityTotalTimeout configures the EgressIP node reachability check timeout.
+	// The default is 1 second. A value of 0 disables the reachability check.
+	// +kubebuilder:default=1
+	// +optional
```

Comment:
You may also want to look at turning this into a `metav1.Duration` which would give you the option to allow users to specify their own units (this is used for a lot of other timeouts)

Reply:
0 was already used as ovnk default config, not sure I understand ur concern with json marshaller I can convert back and forth from json to go 
https://go.dev/play/p/wIv0hWWTB8L

Reply:
You missed the omitempty from your playground example, if you add it in you can see it doesn't show the field in the output, this is the concern. We want the marshalled object to show all of the available options to a user so that they know what they can configure.

My recommendation, based on slack, is to have this be `*metav1.Duration`, where a nil means that this should be defaulted by whichever operator consumes this, and then an explicit `0` value means to disable the timeout as you've mentioned before.

Reply:
If you feel strongly about not using a duration, I think this would then have to be `*int32`, though we have a lot of other timeouts as durations so it would be better to be consistent

Reply:
since this really multiplier of how many seconds that I think we are better of with `*int32` it make the code much cleaner, I did unit-test the code and the new api with CNO and its clean

Reply:
Did you apply the updated schema to a cluster and manually test that you can update the field, delete the field, does it disappear when deleted?

Reply:
I did add unit-test cases for egip new config I will run it via the new schema on life cluster

Reply:
here is the unit-test with CNO on life cluster
```
- scale down CVO
oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
- edit crd to add the new field
oc edit crd networks.operator.openshift.io
and make sure its there 
oc get crd networks.operator.openshift.io -o yaml | grep egressIPConfig


- change EIP total timeout to 5sec
spec:
  clusterNetwork:
  - cidr: 10.128.0.0/14
    hostPrefix: 23
  defaultNetwork:
    ovnKubernetesConfig:
      egressIPConfig:
        reachabilityTotalTimeoutSeconds: 5
      gatewayConfig:
        routingViaHost: false
      genevePort: 6081
      mtu: 8901
      policyAuditConfig:
        destination: "null"
        maxFileSize: 50
        rateLimit: 20
        syslogFacility: local0
    type: OVNKubernetes
  deployKubeProxy: false
  disableMultiNetwork: false
  disableNetworkDiagnostics: false

- change to 0
spec:
  clusterNetwork:
  - cidr: 10.128.0.0/14
    hostPrefix: 23
  defaultNetwork:
    ovnKubernetesConfig:
      egressIPConfig:
        reachabilityTotalTimeoutSeconds: 0
      gatewayConfig:
        routingViaHost: false
      genevePort: 6081
      mtu: 8901
      policyAuditConfig:
        destination: "null"
        maxFileSize: 50
        rateLimit: 20
        syslogFacility: local0

- delete config
spec:
  clusterNetwork:
  - cidr: 10.128.0.0/14
    hostPrefix: 23
  defaultNetwork:
    ovnKubernetesConfig:
      gatewayConfig:
        routingViaHost: false
      genevePort: 6081
      mtu: 8901
      policyAuditConfig:
        destination: "null"
        maxFileSize: 50
        rateLimit: 20
        syslogFacility: local0

- add again with value of 3
spec:
  clusterNetwork:
  - cidr: 10.128.0.0/14
    hostPrefix: 23
  defaultNetwork:
    ovnKubernetesConfig:
      egressIPConfig:
        reachabilityTotalTimeoutSeconds: 3
      gatewayConfig:
        routingViaHost: false
      genevePort: 6081
      mtu: 8901
      policyAuditConfig:
        destination: "null"
        maxFileSize: 50
        rateLimit: 20
        syslogFacility: local0
    type: OVNKubernetes
```

Reply:
Thanks for testing this out